drivers/ethernet/eth_mcux.c

```diff
@@ -930,6 +944,46 @@ static void eth_callback(ENET_Type *base, enet_handle_t *handle,
 	}
 }
 
+static void eth_rx_thread(void *arg1, void *unused1, void *unused2)
```

Comment:
Would it be possible to split this commit into a few separate ones?
For example, something like this:
- commit moving Ethernet RX/TX handlers to threads
- commit that fixes the `ENET_SendFrame error: 4004` issue (if this was a specific, smaller changeset)
- commit with remaining PTP specific changes

Feel free to divide up this commit however you see fit, but I think it would make this PR much more approachable if the changes to the Ethernet driver were split across commits.

Reply:
Thank you for your suggestion @danieldegrasse . I have done a first attempt at splitting the commit. Have a look, and let me know if you like the new proposal better.